Cabal/Distribution/Utils/Json.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Extremely simple JSON helper. Don't do anything too fancy with this!
+module Distribution.Utils.Json
+    ( Json(..)
+    , (.=)
+    , renderJson
+    ) where
+
+import Data.Text (Text)
+import qualified Data.Text as Text
+
+data Json = JsonArray [Json]
+          | JsonBool !Bool
+          | JsonNull
+          | JsonNumber !Int
+          | JsonObject [(Text, Json)]
+          | JsonRaw !Text
+          | JsonString !Text
+
+-- | A type to mirror 'ShowS'
+type ShowT = Text -> Text
```

Comment:
I spotted this via your comment. Why this and not https://hackage.haskell.org/package/text-1.2.4.0/docs/Data-Text-Lazy-Builder.html

The `ShowT = Text -> Text` is worse than `ShowS`, as it copies all intermediate values, causing quadratic behavior.

Yet, if you doing this for performance reasons, you should use `ByteString` and its builder, like `aeson`.

Reply:
Accidentally quadratic, how embarrassing! I'll switch this all over to `ByteString`, didn't realise `Text` was also locale dependent